src/pieces/commandManager.ts

```diff
@@ -69,54 +89,95 @@ async function loadCommands(bot: Client) {
-		[commandName] = msg.content.slice(PREFIX.length).trim().split(' ');
-	} else {
-		[commandName] = msg.content.split(' ');
-	}
-	const unparsedArgs = msg.content.slice(msg.content.indexOf(commandName) + commandName.length, msg.content.length).trim();
+		const botCmd = bot.commands.find(cmd => cmd.name === command.name);
+		if (botCmd
+			&& (botCmd.tempPermissions.length !== curPerms.length
+			|| !botCmd.tempPermissions.every(perm =>
+				curPerms.find(curPerm => isPermissionEqual(curPerm, perm))))) {
+			console.log(`Updating permissions for ${botCmd.name}`);
+			permsUpdated++;
+			return commands.permissions.set({
+				command: command.id,
+				permissions: botCmd.tempPermissions
+			});
+		}
+	}));
 
-	const command = getCommand(msg.client, commandName);
-	if (!command || command.enabled === false) return;
+	console.log(`${bot.commands.size} commands loaded (${numNew} new, ${numEdited} edited) and ${permsUpdated} permission${permsUpdated === 1 ? '' : 's'} updated.`);
+}
 
-	if (msg.channel.type === 'DM' && command.runInDM === false) return msg.reply(`${command.name} is not available in DMs.`);
-	if (msg.channel.type === 'GUILD_TEXT' && command.runInGuild === false) {
-		await msg.author.send(`<@!${msg.author.id}>, the command you just tried to run is not available in public channels. Try again in DMs.`)
-			.catch(async () => { await msg.reply('That command is not available here, try again in DMs'); });
-		return msg.delete();
+async function runCommand(interaction: CommandInteraction, bot: Client): Promise<unknown> {
+	const command = bot.commands.get(interaction.commandName);
+	if (interaction.channel.type === 'DM' && command.runInDM === false) {
+		return interaction.reply('This command cannot be run in DMs');
 	}
 
-	if (command.permissions && !await command.permissions(msg)) return msg.reply('Missing permissions');
-
-	let args: Array<unknown>;
-	if (command.argParser) {
-		try {
-			args = await command.argParser(msg, unparsedArgs);
-		} catch (error) {
-			msg.channel.send(error);
-			return;
-		}
-	} else {
-		args = [unparsedArgs];
+	if (interaction.channel.type === 'GUILD_TEXT' && command.runInGuild === false) {
+		return interaction.reply({
+			content: 'This command must be run in DMs, not public channels',
+			ephemeral: true
+		});
 	}
 
-	try {
-		command.run(msg, args)
-			?.catch(async (error: Error) => {
-				msg.reply(`An error occurred. ${MAINTAINERS} have been notified.`);
-				msg.client.emit('error', new CommandError(error, msg));
-			});
-	} catch (error) {
-		msg.reply(`An error occurred. ${MAINTAINERS} have been notified.`);
-		msg.client.emit('error', new CommandError(error, msg));
-	}
+	if (bot.commands.get(interaction.commandName).tempRun !== undefined) return bot.commands.get(interaction.commandName)?.tempRun(interaction);
+	else return interaction.reply('We haven\'t switched that one over yet');
+	// interaction.reply(interaction.commandName);
```

Comment:
This comment block should be removed if we no longer need to refer to it

Reply:
Will do during the big yeet:tm: